### DISCUSSION TITLE
How to type Tooltip's `placement` prop in Typescript?

### DISCUSSION THREAD

**Discussion by christiankaindl** [ariakit/ariakit#860] [Questions] [answered]
Hi there 👋! I am trying to build a high-level Tooltip component, which allows users to pass a `placement` prop which is passed on to Reakit.

Everything is working fine, but I'm having trouble typing the `placement` prop. Reakit [doesn't export the type explicitly](https://github.com/reakit/reakit/blob/eec931d8b4210f31bc562e60392d73f5047109d0/packages/reakit/src/Popover/PopoverState.ts#L20), so I tried to create it myself (see code example below) by using Pick on existing Reakit types, but without luck. (Note: The component code is mostly from the ["Abstracting" section in the docs](https://reakit.io/docs/tooltip/#abstracting)):

```tsx
import { cloneElement, forwardRef, ReactElement, ReactNode } from "react"
import {
  TooltipReference,
  Tooltip as BaseTooltip,
  useTooltipState, 
  TooltipInitialState
} from "reakit"

interface Props {
  content: string | ReactNode
  placement?: Pick<TooltipInitialState, 'placement'>
  children: ReactElement
}

const Tooltip = forwardRef<HTMLDivElement, Props>(function ({
  content,
  placement,
  children
}, ref) {
  const tooltipState = useTooltipState({
    placement
  })
  return (
    <>
      <TooltipReference {...children.props} {...tooltipState} ref={ref}>
        {referenceProps => cloneElement(children, referenceProps)}
      </TooltipReference>
      <BaseTooltip {...tooltipState}>
        {content}
      </BaseTooltip>
    </>
  )
})
export default Tooltip
```

This gives me the following TS error:
```
Type 'Pick<TooltipInitialState, "placement"> | undefined' is not assignable to type 'Placement | undefined'.
```
Should I just copy/duplicate the type from the source code or is there a different solution for typing the prop?

**Comment by diegohaz** [comment, marked as answer]
`Pick<TooltipInitialState, "placement">` will result in an object with a `placement` property. What you want is `TooltipInitialState["placement"]`.

**Reply by christiankaindl** [reply]
Yes, thanks so much. Learned something about TypeScript today 😄 🙌
